### PR TITLE
Toyota: gate pedal behind long

### PR DESCRIPTION
**Pull request by cydia2020**

### CODE REVIEW

**Comment by pd0wm** [comment]
Any specific reason for this change?

**Comment by cydia2020** [comment]
From what I can tell, if someone has their DSU plugged in with a pedal (this scenario is very rare but there is **nothing** to stop users from doing that), openpilot will still send pedal commands without a clear way to decelerate if the stock ACC isn't detecting the lead car. This feels like a potential safety hazard.

**Comment by pd0wm** [comment]
Maybe it makes more sense to check the condition at the point where we actually command the gas. The gas interceptor *is* present, so it makes sense that the value in carParams is set.

**Comment by cydia2020** [comment]
> Maybe it makes more sense to check the condition at the point where we actually command the gas. The gas interceptor _is_ present, so it makes sense that the value in carParams is set.

So should this be done from car controller instead?

**Comment by pd0wm** [comment]
Yes, I think that makes the most sense.

File: selfdrive/car/toyota/carcontroller.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def update(self, enabled, active, CS, frame, actuators, pcm_cancel_cmd, hud_aler
              left_line, right_line, lead, left_lane_depart, right_lane_depart):
 
     # gas and brake
-    if CS.CP.enableGasInterceptor and enabled:
+    if CS.CP.enableGasInterceptor and CS.CP.openpilotLongitudinalControl and enabled:
```
Comment (pd0wm) [inline selfdrive/car/toyota/carcontroller.py:29]: We would still send the CAN packets to the interceptor on line 103. Probably best to combine these two blocks.

Comment (pd0wm) [inline selfdrive/car/toyota/carcontroller.py:29]: It's fine to make the `interceptor_gas_cmd`. You want to gate the sending on line 103

**Pull request merged**